legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/api/GitLabEntityApi.java

```diff
@@ -917,9 +917,44 @@ synchronized Entity getEntity()
                 Entity localEntity = this.sourceDirectory.deserialize(this.file);
                 if (!Objects.equals(localEntity.getPath(), getEntityPath()))
                 {
-                    throw new RuntimeException("Expected entity path " + getEntityPath() + ", found " + localEntity.getPath());
+                    /**
+                     *  In gitlab, each entity will be stored as a file of which file name is built upon entity's path.
+                     *  for now, the file name of function is functionName, which doesn't contain function signature.
+                     *  To support function overloading, Engine is updated to store function signature as the name of
+                     *  PackageableElement, which updates PackageableElement's path at the same time since path is
+                     *  built upon PackageableElement's name. Any new function entity will be stored in a file whose
+                     *  name is the full function signature in gitlab. However, existing functions need to be updated
+                     *  to be consistent.
+                     *
+                     *  getEntityPath() is from file name in gitlab
+                     *  localEntity is computed from Engine, once Engine is updated, it will contain the whole function signature.
+                     *  For existing functions in gitlab, localEntity.getPath() and getEntityPath() won't match anymore.
+                     *  However, they are related. localEntity.getPath() must contain getEntityPath() if everything is correct
+                     *  for these cases. If condition below is true, a new entity will be created using the short version
+                     *  of the path (functionName only) as a flag for Studio to trigger a `delete` and `create` actions.
+                     *  Users have to push local changes computed automatically for functions above, which will delete old files
+                     *  and create corresponding new files for functions. Therefore, gitlab will keep clean and consistent.
+                     */
```

Comment:
It needs to be made clear that this is a hack, that it is temporary, and that it should be removed as soon as possible.
```suggestion
                    /**
                     * This is a temporary hack to compensate for a non-backward-compatible change in 
                     * ConcreteFunctionDefinition parsing in legend-engine. Previously, the function name was used as the
                     * PackageableElement name by the parser. Now, however, a new PackageableElement name is created
                     * based on the function name and signature (i.e., the same name the Pure compiler would give it).
                     *
                     * Normally, an exception would be thrown when there is a mismatch between the expected entity path
                     * (based on the file path) and the one computed from the parsed entity (based on the "package" and
                     * "name" properties). However, to compensate for this non-backward-compatible change, if the entity is a
                     * ConcreteFunctionDefinition, the entity path and "name" property will be changed to match the expected
                     * values.
                     *
                     * This hack should be removed at the earliest opportunity, and no later than 3/31/2023.
                     */
```